src/test/java/com/microsoft/sqlserver/jdbc/exception/ExceptionTest.java

```diff
@@ -93,6 +95,50 @@ public void testSocketTimeoutExceptionCause() throws Exception {
         }
     }
 
+    @Test
+    public void testResultSetErrorSearch() throws Exception {
+        String tableName = AbstractSQLGenerator.escapeIdentifier(RandomUtil.getIdentifier("ISSUE659TABLE"));
+        String procName = AbstractSQLGenerator.escapeIdentifier(RandomUtil.getIdentifier("ISSUE659PROC"));
+        String expectedException = "Error occured during the insert";
+        int outputValue = 5;
+        String createTableSql = "CREATE TABLE " + tableName + "(FIELD1 VARCHAR (255) NOT NULL);";
+        String createProcSql = "CREATE PROCEDURE " + procName
+                + " AS BEGIN TRANSACTION\n BEGIN TRY SET NOCOUNT ON; INSERT INTO " + tableName + " (FIELD1) OUTPUT "
+                + outputValue + " VALUES ('test'); INSERT INTO" + tableName
+                + " (FIELD1) VALUES (NULL); COMMIT TRANSACTION; END TRY BEGIN CATCH DECLARE @errorMessage NVARCHAR(4000) = ERROR_MESSAGE(); ROLLBACK TRANSACTION; RAISERROR('"
+                + expectedException + ": %s', 16, 1, @errorMessage); END CATCH;";
+        String execProcSql = "EXECUTE " + procName;
+
+        try (Connection conn = PrepUtil.getConnection(connectionString); Statement stmt = conn.createStatement()) {
+            TestUtils.dropTableIfExists(tableName, stmt);
+            TestUtils.dropProcedureIfExists(procName, stmt);
+            stmt.execute(createTableSql);
+            stmt.execute(createProcSql);
+            stmt.execute(execProcSql);
+            try (ResultSet rs = stmt.getResultSet()) {
+                // First result set
+                while (rs.next()) {
+                    assertEquals(outputValue, rs.getInt(1));
+                }
+            } catch (SQLException e) {
+                // First result set should not throw an exception.
+                fail();
+            }
+            try {
+                // Second result set, contains the exception.
+                assertTrue(stmt.getMoreResults());
+                fail();
```

Comment:
fail with some message (Exception should not have thrown - available in Test Resources)